dev/com.ibm.ws.microprofile.config.1.3/src/com/ibm/ws/microprofile/config13/sources/AppPropertyConfigSource.java

```diff
@@ -92,14 +94,17 @@ public AppPropertyConfigSource() {
 
     private SortedSet<Configuration> getOSGiConfigurations() {
         PrivilegedAction<SortedSet<Configuration>> configAction = () -> {
+            SortedSet<Configuration> osgiConfigs = null;
             if (bundleContext == null) {
                 bundleContext = OSGiConfigUtils.getBundleContext(getClass());
             }
-            if (applicationName == null) {
-                applicationName = OSGiConfigUtils.getApplicationName(bundleContext);
-            }
-            SortedSet<Configuration> osgiConfigs = OSGiConfigUtils.getConfigurations(bundleContext, applicationName);
+            if (bundleContext != null) { //bundleContext could be null if not inside an OSGi framework, e.g. unit test
+                if (applicationName == null) {
+                    applicationName = OSGiConfigUtils.getApplicationName(bundleContext);
```

Comment:
I don't really like this sort of thing, would be nicer if we could refactor to allow unit tests without having all this special casing.

Reply:
i don't really want to fake up an OSGi BundleContext